lib/hipe/test/basic_SUITE_data/basic_tuples.erl

```diff
@@ -73,6 +75,16 @@ get_elements([Element|Rest], Tuple, Pos) ->
 get_elements([], _Tuple, _Pos) ->
   ok.
 
+squares() ->
+  {1*1,   2*2,   3*3,   4*4,   5*5,   6*6,   7*7,   8*8,   9*9,   10*10,
+   11*11, 12*12, 13*13, 14*14, 15*15, 16*16, 17*17, 18*18, 19*19, 20*20}.
+
+test_square() ->
+  %% A bit of gymnastics to create a variable with a 1..19 range
```

Comment:
The purpose of the test is to exercise the edge case in which the patch causes the `element/2` bounds test to be omitted where it was not before. As for the comment on the variable's range, I was writing about what the range analysis will think (since it cannot assume anything about `rand:uniform/1`). I will rewrite it to be more clear about that as well as explicitly say what edge case it is aiming for.

As for why the test broke the pattern, passing variables with these kinds of ranges tended to widen them to something like `0..256`, which is why I chose the easy solution of putting the entire test in one function.

I will rewrite the test to fit the pattern better, drop the determinism, and be slightly less contrived.